### ISSUE TITLE
Wrong statVersion passed to __fxstat64

### ISSUE THREAD

**Issue by nyh** [jnr/jnr-posix#16]
The code in LinuxPOSIX.java passed version "0" for 64 bit x86 Linux. But I believe this is an error, and needs to be 1 (see _STAT_VER in glibc's sysdeps/unix/sysv/linux/x86/bits/stat.h, or just compile a test program and print out _STAT_VER).

This doesn't cause any problems because glibc's __fxstat64 actually never checks the version argument, but it nevertheless seems strange to give it the wrong version.

Another related question - it seems you always call __fxstat64, even on 32-bit platforms. So why do you need that 32-bit structure definition?

**Comment by headius** [comment]
Closing due to age; if this is still valid, please assist us in finishing it up.

**Comment by nyh** [comment]
I'm not sure why you're closing this issue "due to age", I think it should have been rightly closed because it was fixed! I looked at the current code now, and I see that it has code which tries to stat a must-exist file with version=0, and if that fails, it will use version=1. Seems a reasonable fix.

By the way, I'm curious why you used "/dev/null" as an example of a must-exist file in LinuxPosix's constructor. I know that "/dev/null" does exist almost universally on Linux, but must it really really always exist? Can't some esoteric Linux distribution or Linux variant function without it? I think it would have been even safer to use "/" in the test! Surely, "/" must always, always, exist :-) Anyway, this is just a minor nitpick, and doesn't prevent this issue from being closed.
